starter-code/src/main.js

```diff
@@ -25,23 +25,185 @@ var cards = [
+   */
+  addOnClickListenners(){
+    this.gameDivMapping.forEach(cardELement => {
+      cardELement['div'].addEventListener(
+        "click",
+          e => this.onClickEventListenner(e, cardELement)
+      );
+    })
+  }
+
+  /**
+   * Event listenner on one div
+   * @param e: event to handle
+   * @param cardElement: object witj keys 'div' and 'card'. This should be an element belonging to the array this.gameDivMapping
+   */
+  onClickEventListenner(e, cardElement){
+
+    // Do nothing if two or more cards are returned
+    if (this.cardsCurrentlyOnPlay.length < 2){
+
+      // 0. do not do anything if one card is already returned, and we are trying to click it again
+      if( this.cardsCurrentlyOnPlay.length === 1 &&
+          this .cardsCurrentlyOnPlay[0].index === cardElement.index
+      ){
+        return
+      }
+
+      // 1. return card on the dom
+      this.returnCardOnTheBoard(cardElement['div']);
+
+      // 2. update the list of returned cards
+      this.cardsCurrentlyOnPlay.push(cardElement);
+
+      // 3. Only act if two cards are returned on the board, otherwise do nothing
+      if( this.cardsCurrentlyOnPlay.length === 2 ){
+        this.handleTwoCardsFront();
+      }
+    }
+  }
+
+  /**
+   * What happens when two CARDS are on the front
+   */
+  handleTwoCardsFront(){
+
+    // check if the returned cards are a pair
+    // N.B : this also updates the properties this.game.pairsClicked and this.game.pairsGuessed
+    var isPair =  this.game.checkIfPair(
+      this.cardsCurrentlyOnPlay[0]["card"],
+      this.cardsCurrentlyOnPlay[1]["card"]
+    );
+
+    // update the score on the dom
+    $("#pairs_clicked").html(this.game.pairsClicked);
+    $("#pairs_guessed").html(this.game.pairsGuessed);
+
+
+    // if it is a pair, update state game
+
+    if(isPair){
+      this.handlePairFound();
+    }
+    else{
+      // Time out is here to give to the player a reaction time to remember the cards
+      setTimeout(()=>this.handlePairNotFound(), WAITING_TIME_MS);
+    }
+  }
+
+  /**
+   * What happens when a selected pair os the correct one ?
+   */
+  handlePairFound(){
+
+    // 2. update the internal list of returned CARDS
+    this.cardsCurrentlyOnPlay = [];
+
+    if(this.game.isFinished()){
+      console.log("GAME FINISHED !")
+      //TODO: add here something that inform the user that :
+      // - the game ended
+      // - recap the score
+      // - give option to replay
+    }
+  }
+
+  /**
+   * What happens when a selected pair os the wrong one ?
+   */
+  handlePairNotFound(){
+
+    // 1. return back the two CARDS
+    this.cardsCurrentlyOnPlay.forEach(
+      cardElement => {
+        this.returnCardOnTheBoard(cardElement["div"]);
+      }
+    );
+
+    // 2. update the internal list of returned CARDS
+    this.cardsCurrentlyOnPlay = [];
+  }
+
+  /**
+   * Takes as an input a div card and return it on the board (dom manipulation)
+   * @param div
+   */
+  returnCardOnTheBoard(div){
```

Comment:
Try to avoid ambiguous words like `return` that already have a different meaning in code. Maybe this function could be named `resetCard` or something like that.